src/matlab2tikz.m

```diff
@@ -4825,8 +4825,48 @@ function warnFacetedInterp(m2t)
             cbarStyleOptions = opts_add(cbarStyleOptions,...
                 'xticklabel pos', 'upper');
         case 'southoutside'
-
             cbarTemplate = 'horizontal';
+        case 'manual'
+            origUnits = handle.Units;
+            origAxesUnits = handle.Axes.Units;
+            handle.Units = 'centimeters';        % We need absolute
+            handle.Axes.Units = 'centimeters';   % positions for tikz
+            cbarPos = handle.Position;
+            cbarAxesPos = handle.Axes.Position;
+            handle.Units = origUnits;
+            handle.Axes.Units = origAxesUnits;
+
+            % Cases of colorbar axis locations (in or out) depending on position
+            % of colorbar relative to it's associated axes.
+            % According to matlab manual (R2016a) colorbars with Location 'manual'
+            % can only be vertical.
+            if cbarPos(1) + 0.5*cbarPos(3) < cbarAxesPos(1) + 0.5*cbarAxesPos(3) 
+                if strcmp(handle.AxisLocation,'in')
+                    cbarTemplate = 'right';
+                    cbarStyleOptions = opts_add(cbarStyleOptions, 'anchor',...
+                        'south east');
+                else
+                    cbarTemplate = 'left';
+                    cbarStyleOptions = opts_add(cbarStyleOptions, 'anchor',...
+                        'south west');
+                end
+            else
+                if strcmp(handle.AxisLocation,'in')
+                    cbarTemplate = 'left';
+                    cbarStyleOptions = opts_add(cbarStyleOptions, 'anchor',...
+                        'south west');
+                else
+                    cbarTemplate = 'right';
+                    cbarStyleOptions = opts_add(cbarStyleOptions, 'anchor',...
+                        'south east');
+                end
+            end
+            
+            cbarStyleOptions = opts_add(cbarStyleOptions, 'at',...
```

Comment:
Please have a look at the `formatDim` function to print the coordinates:
`formatDim(cbarPos(...), 'cm')` instead of `sprintf('%fcm', cbarPos(...))`.

That way, we have uniform control over how dimensions/coordinates are displayed and what precision is used.